### PR TITLE
update ingress controller to 0.9.0-beta.2 and add prometheus scrape annotations to service

### PR DESCRIPTION
**Pull request by auhlig**
***WIP - DO NOT MERGE***

Does it make sense to try [0.9.0-beta.1](https://github.com/kubernetes/ingress/blob/master/controllers/nginx/Changelog.md#09-beta1) of the ingress controller as this version includes https://github.com/kubernetes/ingress/pull/36,
@databus23?
Would test in staging after lunch if you agree.

### CODE REVIEW

File: system/kube-system/charts/ingress/templates/ingress-controller-daemonset.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ spec:
     spec:
       terminationGracePeriodSeconds: 60
       containers:
-      - image: gcr.io/google_containers/nginx-ingress-controller:0.8.3
+      - image: gcr.io/google_containers/nginx-ingress-controller:0.9.0-beta.1
```
Comment (databus23) [inline system/kube-system/charts/ingress/templates/ingress-controller-daemonset.yaml]: Why not 0.9.0-beta.2?

File: system/kube-system/charts/ingress/templates/ingress-service.yaml [hunk 2]
```diff
@@ -2,8 +2,10 @@ kind: Service
 apiVersion: v1
 
 metadata:
-  name: ingress-controller 
-
+  name: ingress-controller
+  annotations:
+      prometheus.io/scrape: "true"
+      prometheus.io/port: "9101"
```
Comment (MaikuMori) [inline system/kube-system/charts/ingress/templates/ingress-service.yaml]: Why this port? Looking at the sources `/metrics` should be on the same port as `/healthz` which is `10254`.

**Comment by MaikuMori** [comment]
Beta.1 has serious bug where it fails to work with services which use port name instead of port as int.

You also need to adopt other changes before merge such as `--nginx-configmap` is now `--configmap`.

File: system/kube-system/charts/ingress/templates/ingress-service.yaml (hunk 2 above)
Comment (databus23) [inline system/kube-system/charts/ingress/templates/ingress-service.yaml]: @MaikuMori: We are experimenting with a custom `nginx.tpl` that contains instrumentation using https://github.com/knyar/nginx-lua-prometheus providing requests metrics.

This is not in a public repository yet so this is confusing. Apologies for that.

@auhlig Maybe we should switch the service annotation to the metrics port of the nginx-controller and add another annotation to the pod template for our custom metrics.

File: system/kube-system/charts/ingress/templates/ingress-controller-daemonset.yaml [hunk 3]
```diff
@@ -41,7 +44,9 @@ spec:
             containerPort: 80
           - name: https
             containerPort: 443
+          - name: metrics
+            containerPort: 10254
```
Comment (databus23) [inline system/kube-system/charts/ingress/templates/ingress-controller-daemonset.yaml]: I think we are missing the port for the request metrics here:

File: system/kube-system/charts/ingress/templates/ingress-service.yaml [hunk 4]
```diff
@@ -14,6 +16,8 @@ spec:
       targetPort: http
     - name: https 
       port: 443 
-      targetPort: https 
+      targetPort: https
+    - name: metrics
```
Comment (databus23) [inline system/kube-system/charts/ingress/templates/ingress-service.yaml]: I think this is not needed here. prometheus scrapes the endpoints directly

**Comment by databus23** [comment]
Applied manually in staging. LGTM

**Pull request merged**